fix: exclude service worker from tsconfig again (#11727)
sveltejs/kit: documentation/docs/30-advanced/40-service-workers.md

const sw = self as unknown as ServiceWorkerGlobalScope;
```

> This disables access to DOM typings like `HTMLElement` which are not available inside a service worker and instantiates the correct globals. The reassignment of `self` to `sw` allows you to type cast it in the process (there are a couple of ways to do this, but the easiest that requires no additional files). Use `sw` instead of `self` in the rest of the file. The reference to the SvelteKit types ensures that the `$service-worker` import has proper type definitions. If you import `$env/static/public` you either have to `// @ts-ignore` the import or add `/// <reference types="../.svelte-kit/ambient.d.ts" />` to the reference types.

Reviewer: should we be recommending `@ts-ignore` when there's a real solution?
```suggestion
This disables access to DOM typings like `HTMLElement` which are not available inside a service worker and instantiates the correct globals. The reassignment of `self` to `sw` allows you to type cast it in the process (there are a couple of ways to do this, but the easiest that requires no additional files). Use `sw` instead of `self` in the rest of the file. The reference to the SvelteKit types ensures that the `$service-worker` import has proper type definitions. If you import `$env/static/public` you will have to add `/// <reference types="../.svelte-kit/ambient.d.ts" />` to the reference types.
```

Author: I mentioned both because of my hesitancy to make everyone add the `.svelte-kit` path, which kinda sucks - but I also don't see a better way.

Reviewer: Yeah, I don't love adding `.svelte-kit` either, but it seems better than `@ts-ignore`. And probably better to have one ugly thing in the docs than two :laughing:

Author: Yeah you're right. How about
```suggestion
This disables access to DOM typings like `HTMLElement` which are not available inside a service worker and instantiates the correct globals. The reassignment of `self` to `sw` allows you to type cast it in the process (there are a couple of ways to do this, but this is the easiest that requires no additional files). Use `sw` instead of `self` in the rest of the file. The reference to the SvelteKit types ensures that the `$service-worker` import has proper type definitions. If you import `$env/static/public` you will have to add `/// <reference types="../.svelte-kit/ambient.d.ts" />` to the reference types.

These additional steps are necessary to prevent the service worker types from bleeding over into the rest of your application.
```

Author: Coming back to this, I wanna keep the `@ts-ignore` option, the reference types thing really sucks.

> This disables access to DOM typings like `HTMLElement` which are not available inside a service worker and instantiates the correct globals. The reassignment of `self` to `sw` allows you to type cast it in the process (there are a couple of ways to do this, but the easiest that requires no additional files). Use `sw` instead of `self` in the rest of the file. The reference to the SvelteKit types ensures that the `$service-worker` import has proper type definitions. If you import `$env/static/public` you either have to `// @ts-ignore` the import or add `/// <reference types="../.svelte-kit/ambient.d.ts" />` to the reference types.

Reviewer: ```suggestion
This disables access to DOM typings like `HTMLElement` which are not available inside a service worker and instantiates the correct globals. The reassignment of `self` to `sw` allows you to type cast it in the process (there are a couple of ways to do this, but this is the easiest that requires no additional files). Use `sw` instead of `self` in the rest of the file. The reference to the SvelteKit types ensures that the `$service-worker` import has proper type definitions. If you import `$env/static/public` you either have to `// @ts-ignore` the import or add `/// <reference types="../.svelte-kit/ambient.d.ts" />` to the reference types.
```
